### PR TITLE
exec: report -1 (255) on signal exit

### PR DESCRIPTION
**Pull request by brauner**
This makes exec behave even more similar to ssh.

Signed-off-by: Christian Brauner <christian.brauner@ubuntu.com>

### CODE REVIEW

**Comment by brauner** [comment]
Closes https://github.com/lxc/lxd/issues/2774.

**Comment by brauner** [comment]
As discussed, current testsuite failure is expected.

File: test/suites/basic.sh [hunk 1]
```diff
@@ -404,7 +404,7 @@ test_basic_usage() {
   lxc launch testimage foo -e
 
   OLD_INIT=$(lxc info foo | grep ^Pid)
-  lxc exec foo reboot
+  lxc restart foo
```
Comment (stgraber) [inline test/suites/basic.sh]: So this particular test is specifically for container-initiated reboot, so we should keep it using exec.

Comment (brauner) [inline test/suites/basic.sh]: Addressed.

**Pull request merged**